frontend/src/components/NavDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+import {
+  Dispatch,
+  JSX,
+  SetStateAction,
+  useEffect,
+} from "react";
+
+import clsx from "clsx";
+import { noop } from "lodash";
+import { IndexType } from "src/types/generalTypes";
+
+import {
+  Menu,
+  NavDropDownButton,
+} from "@trussworks/react-uswds";
+
+interface NavDropdownProps {
+  activeNavDropdownIndex: IndexType;
+  index: number;
+  isCurrent: boolean;
+  linkText: string;
+  menuItems: JSX.Element[];
+  setActiveNavDropdownIndex: Dispatch<SetStateAction<IndexType>>;
+}
+
+export default function NavDropdown({
+  activeNavDropdownIndex,
+  index,
+  isCurrent,
+  linkText,
+  menuItems,
+  setActiveNavDropdownIndex,
+}: NavDropdownProps): JSX.Element {
+  let isOpen = activeNavDropdownIndex === index;
+
+  function handleToggle() {
+    const activeIndex: IndexType = isOpen ? null : index;
+    setActiveNavDropdownIndex(activeIndex);
+  }
+
+  useEffect(() => {
+    isOpen = activeNavDropdownIndex === index;
+  }, [activeNavDropdownIndex, index]);
```

Comment:
when we were discussing I had a feeling I was forgetting something, and remembered later - rather than the defining the variable and changing it in the useEffect, you could just assign the variable in a useMemo. This is good for now, but something for later maybe.

Reply:
@doug-s-nava we should discuss in morning!